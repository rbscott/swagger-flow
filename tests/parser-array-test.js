import yaml from 'js-yaml';

import parse from '../src/parser';

describe('parse', () => {
  describe('array', () => {
    it('Should parse a basic array', () => {
      const input = yaml.safeLoad(`
definitions:
  basicArray:
    type: array
    items:
      type: string
`);
      expect(parse(input).items).toMatchSnapshot();
    });

    it('Should parse arrays of arrays.', () => {
      const input = yaml.safeLoad(`
definitions:
  basicArray:
    type: array
    items:
      type: array
      items:
        type: string
`);
      expect(parse(input).items).toMatchSnapshot();
    });

    it('Should parse arrays of objects.', () => {
      const input = yaml.safeLoad(`
definitions:
  basicArray:
    type: array
    items:
      type: object
      properties:
        field1:
          type: string
`);
      expect(parse(input).items).toMatchSnapshot();
    });
  });
});
