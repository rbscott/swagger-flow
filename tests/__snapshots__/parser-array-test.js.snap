// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse array Should parse a basic array 1`] = `
Array [
  Object {
    "id": "definitions/basicArray",
    "items": Object {
      "id": "definitions/basicArray/ArrayType",
      "name": "basicArray_ArrayType",
      "type": "string",
    },
    "name": "basicArray",
    "type": "array",
  },
]
`;

exports[`parse array Should parse arrays of arrays. 1`] = `
Array [
  Object {
    "id": "definitions/basicArray",
    "items": Object {
      "id": "definitions/basicArray/ArrayType",
      "items": Object {
        "id": "definitions/basicArray/ArrayType/ArrayType",
        "name": "basicArray_ArrayType_ArrayType",
        "type": "string",
      },
      "name": "basicArray_ArrayType",
      "type": "array",
    },
    "name": "basicArray",
    "type": "array",
  },
]
`;

exports[`parse array Should parse arrays of objects. 1`] = `
Array [
  Object {
    "id": "definitions/basicArray",
    "items": Object {
      "additionalProperties": undefined,
      "fields": Array [
        Array [
          "field1",
          Object {
            "id": "definitions/basicArray/ArrayType/Field_field1",
            "name": "basicArray_ArrayType_Field_field1",
            "type": "string",
          },
        ],
      ],
      "id": "definitions/basicArray/ArrayType",
      "name": "basicArray_ArrayType",
      "required": Array [],
      "type": "object",
    },
    "name": "basicArray",
    "type": "array",
  },
]
`;
